tests/unit/charms/lte_core_interface/v0/dummy_provider_charm/src/charm.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+from ops.charm import CharmBase, RelationJoinedEvent
+from ops.main import main
+
+from lib.charms.lte_core_interface.v0.lte_core_interface import CoreProvides
+
+
+class DummyCoreProviderCharm(CharmBase):
+    def __init__(self, *args):
+        """Charm the service."""
+        super().__init__(*args)
+        self.core_provider = CoreProvides(self, "lte-core")
+        self.framework.observe(self.on.lte_core_relation_joined, self._on_lte_core_relation_joined)
+
+    def _on_lte_core_relation_joined(self, event: RelationJoinedEvent):
+        # TODO: do we need leader check?
```

Comment:
I'd say yes. Also if we do this check, we should probably defer the event if it's not picked up by the leader, shouldn't we?

Reply:
Agreed, fixed.